effectful-th/src/Effectful/TH.hs

```diff
@@ -0,0 +1,238 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE TemplateHaskell #-}
+-- | Template Haskell utilities.
+module Effectful.TH
+  ( -- * Generate functions for dynamic effects
+    makeSendFunctions
```

Comment:
Naming is hard, I'm wondering whether it should be `makeEffect` as in other libraries. The problem is of course that it's only for dynamic dispatch.

Reply:
Yeah, I am not really happy with the naming. `makeEffect` is nice; I will change that.